### ISSUE TITLE
Tentative array definition warning has no warning name

### ISSUE THREAD

**Issue by bevin-hansson** [llvm/llvm-project#87766]
If you define an array with an incomplete type, you get a warning about it being a tentative array definition:
```
int a[];
```
```
<source>:12:5: warning: tentative array definition assumed to have one element
   12 | int a[];
      |     ^
```
However, this warning has no name. Even if the tentative array behavior is desired, it's not possible to disable it selectively via the command line or using diagnostics pragmas. This would be useful.

**Comment by shafik** [comment]
The diagnostic is `warn_tentative_incomplete_array` 

CC @AaronBallman is there a reason why we don't give this a name?

**Comment by AaronBallman** [comment]
GCC has several diagnostics with no warning group either, and so I think we just followed their lead. That said, I think this should have a warning group even if GCC doesn't put it under one. I think `-Wtentative-definition-array` is a reasonable enough name for the group.
